cdap-ui/app/cdap/components/Home/index.js

```diff
@@ -108,18 +114,98 @@ class Home extends Component {
       //FIXME: Must redirect if passed invalid namespace
     }
   }
+
+  processQueryString(queryString) {
+    let sortQueryIndex = queryString.indexOf('sort=');
+    let filterIndex = queryString.indexOf('filter=');
+    let searchTermIndex = queryString.indexOf('search=');
+    let filtersArr = [];
+    let sortOpt;
+    let sortQuery;
+    let searchTerm;
+
+    //Set default sort settings
+    sortOpt = this.sortOptions[0];
+
+    //If a sort parameter is provided, parse and set sorting if valid
+    if(sortQueryIndex !== -1){
+      sortQuery = queryString.substring(sortQueryIndex + 5);
+      let endSortIndex = sortQuery.indexOf('&');
+
+      if(endSortIndex !== -1){
+        sortQuery = sortQuery.substring(0, endSortIndex);
+      }
+
+      let finalSortQuery = sortQuery.split('+').join(' ');
+
+      for(let index = 0; index < this.sortOptions.length; index++){
+        if(this.sortOptions[index].sort === finalSortQuery){
+          sortOpt = this.sortOptions[index];
+        }
+      }
+    }
+
+    if(searchTermIndex !== -1){
+      searchTerm = queryString.substring(searchTermIndex  + 7);
+      let endSearchIndex = searchTerm.indexOf('&');
+
+      if(endSearchIndex !== -1){
+        searchTerm = searchTerm.substring(0, endSearchIndex);
+      }
+
+      searchTerm = searchTerm.split('+').join(' ');
+    }
+
+    if(filterIndex !== -1){
+      //Parse url substring ; start after 'filter='
+      let filterString = location.search.substring(filterIndex + 7);
+      filtersArr = filterString.split(',');
+
+      //Process the array and remove any filters that are invalid
+      for(let i = filtersArr.length-1; i >= 0; i--){
```

Comment:
We could use `.filter` here. Should try to use declarative style more than the imperative ones. Less code and easier to understand